torch/_higher_order_ops/cond.py

```diff
@@ -28,16 +30,136 @@
 )
 
 
+@contextmanager
+def _set_compilation_env():
+    _old_is_tracing = torch.fx._symbolic_trace._is_fx_tracing_flag
+    _old_size_limit = torch._dynamo.config.cache_size_limit
+
+    try:
+        # We need to turn off the is_fx_tracing_flag. Remove this flag check from dyanmo
+        # once we are confident fx tracing works with dynamo.
+        torch.fx._symbolic_trace._is_fx_tracing_flag = False
+        # We need to enlarge the dynamo cache size limit because cond's frame is
+        # hitted with different inputs frequently by tests
+        torch._dynamo.config.cache_size_limit = 256
+        yield
```

Comment:
If this is just for testing, then we should just reset() before/after each of the cond tests. Otherwise it is weird for this API to change a config